GitCommands/RevisionReader.cs

```diff
@@ -290,52 +290,50 @@ int CountParents(in ReadOnlySpan<byte> array, int baseOffset)
 
             #region Timestamps
 
-            // Lines 2 and 3 are timestamps, as decimal ASCII seconds since the unix epoch, each terminated by `\n`
-            var authorUnixTime = ParseUnixDateTime(in array);
-            var commitUnixTime = ParseUnixDateTime(in array);
+            // Decimal ASCII seconds since the unix epoch, each terminated by `\n`
+            long authorUnixTime = ParseUnixDateTime(in array);
+            long commitUnixTime = ParseUnixDateTime(in array);
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             long ParseUnixDateTime(in ReadOnlySpan<byte> array)
             {
                 long unixTime = 0;
 
-                while (true)
+                while (offset < array.Length)
                 {
                     int c = array[offset++];
 
                     if (c == '\n')
                     {
-                        return unixTime;
+                        break;
                     }
 
                     unixTime = (unixTime * 10) + (c - '0');
                 }
+
+                return unixTime;
             }
 
             #endregion
 
             #region Encoded string values (names, emails, subject, body)
 
             // Finally, decode the names, email, subject and body strings using the required text encoding
-            // Handle '\v' (Shift-Enter) as '\n' for users that by habit avoid Enter to 'send'
-            // Handle '\v' (Shift-Enter) as '\n' for users that by habit avoid Enter to 'send'
-            ReadOnlySpan<char> s = _logOutputEncoding.GetString(array[offset..]).Replace('\v', '\n').AsSpan();
+            ReadOnlySpan<char> s = _logOutputEncoding.GetString(array[offset..]).AsSpan();
             StringLineReader reader = new(in s);
 
-            var author = reader.ReadLine();
-            var authorEmail = reader.ReadLine();
-            var committer = reader.ReadLine();
-            var committerEmail = reader.ReadLine();
-            var reflogSelector = _hasReflogSelector ? reader.ReadLine() : null;
-
-            bool skipBody = _sixMonths > authorUnixTime;
-            (string? subject, string? body, bool hasMultiLineMessage) = reader.PeekSubjectBody(skipBody);
+            string author = reader.ReadLine();
+            string authorEmail = reader.ReadLine();
+            string committer = reader.ReadLine();
+            string committerEmail = reader.ReadLine();
+            string reflogSelector = _hasReflogSelector ? reader.ReadLine(useStringPool: false) : null;
```

Comment:
It is not obvious to me why (just) in this case the string is not shared.
Ah, the regarding comment is located inside the function. Move it here?

Reply:
Moving the comment, separate for authors etc and reflogselector